site_scons/components/__init__.py

```diff
@@ -135,18 +136,21 @@ def define_mercury(reqs):
                           '--disable-psm3 ' +
                           '--without-gdrcopy ' +
                           OFI_DEBUG +
-                          exclude(reqs, 'psm2',
+                          include(reqs, 'psm2',
                                   '--enable-psm2' +
                                   check(reqs, 'psm2',
                                         "=$PSM2_PREFIX "
                                         'LDFLAGS="-Wl,--enable-new-dtags ' +
                                         '-Wl,-rpath=$PSM2_PREFIX/lib64" ',
                                         ''),
-                                  ''),
+                                  '--disable-psm2 ') +
+                          include(reqs, 'psm3',
```

Comment:
I do wonder if this would actually work, however I expect we'd need more changes to be able to use psm3 anyway.